frontend/src/components/wallet/useWalletProvider.ts

```diff
@@ -81,7 +81,7 @@ export const useWalletProvider = () => {
           method: 'eth_accounts',
         });
 
-        if (result.length > 0) {
+        if (result?.length > 0) {
```

Comment:
Metamask will always return an array here.

Same question as above.  Is this necessary?